src/scalar.rs

```diff
@@ -683,6 +689,65 @@ impl Scalar {
         output
     }
 
+    /// Write this scalar in radix \\(2\^r\\), with \\(n = ceil(256/r)\\) coefficients in \\([-(2\^r)/2,(2\^r)/2)\\),
+    /// i.e., compute \\(a\_i\\) such that
+    /// $$
+    ///    a = a\_0 + a\_1 2\^1r + \cdots + a_{n-1} 2\^{r*(n-1)},
+    /// $$
+    /// with \\(-2\^r/2 \leq a_i < 2\^r/2\\) for \\(0 \leq i < (n-1)\\) and \\(-2\^r/2 \leq a_{n-1} \leq 2\^r/2\\).
+    /// Radix power \\(r\\) can be between 1 and 15.
+    #[cfg(any(feature = "alloc", feature = "std"))]
+    pub(crate) fn to_arbitrary_radix(&self, mut radix_power: usize) -> Vec<i16> {
+        use byteorder::{ByteOrder, LittleEndian};
+        // The last word is half the size, so it always <= 2^r/2 including the carry
+        debug_assert!(self[31] <= 127);
+
+        // Ensure that radix is in [2^1, 2^15]
+        radix_power = if radix_power > 15 { 15 } else { radix_power };
```

Comment:
These should be `debug_assert`s rather than silently changing bad parameters, IMO.

Reply:
Yeah, it makes even more sense if we do not expose these through the API.

Reply:
Done.